kratos/tests/test_skin_detection_process.py

```diff
@@ -79,7 +79,14 @@ def test_NotOnSubModelPartSkinDetectionProcess(self):
         detect_skin.Execute()
 
         # Check the number of conditions created
-        self.assertEqual(self.model_part.GetCommunicator().GlobalNumberOfConditions(), 112)
```

Comment:
Why is this not working? Internally it does exactly the same operation you're doing.

Reply:
I don't know. It says it is zero. I don't know if I need to do something else to update the database

Reply:
If it is mot working then the communicator is also bot properly initialized
Might be related to other problems

Reply:
AS you see, I adapted the test to the same format used in other dual serial/MPI tests

Reply:
I tried with:

~~~sh
# Check the number of conditions created
        if self.model_part.IsDistributed():
            import KratosMultiphysics.mpi as KratosMPI
            KratosMPI.ParallelFillCommunicator(self.model_part, self.model_part.GetCommunicator().GetDataCommunicator()).Execute()
            self.assertEqual(self.model_part.GetCommunicator().GlobalNumberOfConditions(), 112)
        else:
            self.assertEqual(self.model_part.NumberOfConditions(), 112)
~~~

but it fails with more than 3 cores

Reply:
If we avoid this, the PR is ready to merge (compilation fixed)

Reply:
Most likely the comm is the placeholder one for the serial cases, so it just does not have any mLocalMesh initialized.

Reply:
Yes, I tried to define ParallelFillCommunicator in the process, but the compilation fails because is not in the /mpi folder. And moving the process to MPI or separate it between MPI and non MPI is something I want to avoid

Reply:
Are we happy like this?

Reply:
My 2 cents, for me its ok to leave it like this. But we should start considering if its worth to pay the overhead of initializing a communicator with 1 partition for serial cases as well. Some tests could benefit from this and it should be much more intuitive to use (As a user, I would expect that function to work without prior setup)

Reply:
I checked, the communication plan is properly initialized, hence I am not sure why the fct fails

I would investigate it deeper